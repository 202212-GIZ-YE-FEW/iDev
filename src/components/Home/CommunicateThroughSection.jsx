import { withTranslation } from "next-i18next";

import PageIntro from "@/components/PageIntro";

import CommunicateThroughItem from "./CommunicateThroughItem";
function CommunicateThroughSection({ t }) {
    const communicationMeans = [
        {
            name: t("voiceCall"),
            icon: "voice",
            description: t("voiceCallDesc"),
        },
        {
            name: t("chat"),
            icon: "chat",
            description: t("chatDesc"),
        },
        {
            name: t("videoCall"),
            icon: "video",
            description: t("videoCallDesc"),
        },
    ];
    return (
        <>
            <div className='container flex flex-col space-y-20 justify-between'>
                <PageIntro title={t("commThroughTitle")} />
                <div className='grid sm:grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-5'>
                    {communicationMeans.map((item) => {
                        return (
                            <CommunicateThroughItem
                                key={item.name}
                                name={item.name}
                                icon={item.icon}
                                description={item.description}
                            />
                        );
                    })}
                </div>
            </div>
        </>
    );
}
export default withTranslation("home")(CommunicateThroughSection);
