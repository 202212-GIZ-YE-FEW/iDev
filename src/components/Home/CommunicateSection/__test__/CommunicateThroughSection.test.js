import renderer from "react-test-renderer";

import CommunicateThroughSection from "../CommunicateThroughSection";

it("renders correctly", () => {
    const tree = renderer.create(<CommunicateThroughSection />).toJSON();
    expect(tree).toMatchSnapshot();
});
