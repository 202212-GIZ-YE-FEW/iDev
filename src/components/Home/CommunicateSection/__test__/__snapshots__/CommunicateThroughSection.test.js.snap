// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container flex flex-col space-y-20 justify-between"
>
  <div
    className="mb-5"
  >
    <span
      className="font-normal text-3xl md:text-4xl lg:text-5xl block uppercase"
    >
      commThroughTitle
    </span>
  </div>
  <div
    className="grid sm:grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-5"
  >
    <div
      className="bg-light-white flex flex-col items-center justify-center text-center p-9 rounded-3xl space-y-3 drop-shadow-lg"
    >
      <img
        alt="voiceCall svg"
        data-nimg="1"
        decoding="async"
        height={50}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/home/voice.svg"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={50}
      />
      <span
        className="uppercase text-4xl"
      >
        voiceCall
      </span>
      <p
        className="text-sm rtl:text-base text-black/50"
      >
        voiceCallDesc
      </p>
    </div>
    <div
      className="bg-light-white flex flex-col items-center justify-center text-center p-9 rounded-3xl space-y-3 drop-shadow-lg"
    >
      <img
        alt="chat svg"
        data-nimg="1"
        decoding="async"
        height={50}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/home/chat.svg"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={50}
      />
      <span
        className="uppercase text-4xl"
      >
        chat
      </span>
      <p
        className="text-sm rtl:text-base text-black/50"
      >
        chatDesc
      </p>
    </div>
    <div
      className="bg-light-white flex flex-col items-center justify-center text-center p-9 rounded-3xl space-y-3 drop-shadow-lg"
    >
      <img
        alt="videoCall svg"
        data-nimg="1"
        decoding="async"
        height={50}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/home/video.svg"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={50}
      />
      <span
        className="uppercase text-4xl"
      >
        videoCall
      </span>
      <p
        className="text-sm rtl:text-base text-black/50"
      >
        videoCallDesc
      </p>
    </div>
  </div>
</div>
`;
